R/xgx_stat_ci.R

```diff
@@ -101,85 +101,203 @@
     }
     else if (igeom == "line") {
-      if (is.null(temp$aes_params$size)) temp$aes_params$size <- 1
+      if (is.null(lay$aes_params$size)) lay$aes_params$size <- 1
     }
     else if (igeom == "errorbar") {
-      if (is.null(temp$aes_params$size)) temp$aes_params$size <- 1
-      if (is.null(temp$geom_params$width)) {
-        temp$geom_params$width <- 0
-      }
+      if (is.null(lay$aes_params$size)) lay$aes_params$size <- 1
+      if (is.null(lay$geom_params$width)) lay$geom_params$width <- 0
     }
     else if (igeom == "ribbon") {
-      if(is.null(temp$aes_params$alpha)) temp$aes_params$alpha <- 0.25
+      if(is.null(lay$aes_params$alpha)) lay$aes_params$alpha <- 0.25
     }
     else if (igeom == "pointrange") {
-      if(is.null(temp$aes_params$size)) temp$aes_params$size <- 1
-      temp$geom$geom_params$fatten <- 2
+      if(is.null(lay$aes_params$size)){
+        lay$aes_params$size <- 1
+        lay$geom$geom_params$fatten <- 2
+      }
     }
-    
 
-    ret[[paste0("geom_", igeom)]] <- temp
+    lays[[paste0("geom_", igeom)]] <- lay  
   }
-
-  return(ret)
+  
+  return(lays)
 }
+
+
+
+
+
+
+
+#' Stat ggproto object for creating ggplot layers of binned confidence intervals
+#' for probabiliities of classes in ordinal data
+#'
+#' 
+#' \code{StatSummaryBinOrdinal} returns a ggproto object for plotting mean +/- confidence bins
+#' 
+#'
+#' @return ggplot2 ggproto object
+#'
+#' @export
+StatSummaryBinOrdinal <- ggplot2::ggproto("StatSummaryBinOrdinal", ggplot2::Stat,
+
+     required_aes = c("x"),
+     # default_aes = aes(fill = ..y..),
+     
+     compute_group = function(data, scales, conf_level, distribution, bins, breaks,
+                              fun.data = NULL,
+                              fun.args = list()) {
+       return(data)
+     },
+
+     setup_data = function(data, params) {
+
+       # Calculate percentages for each category across each bin
+       # Get median x value for each bin
+       median_x <- data %>% mutate(quantile_index = dplyr::ntile(data$x, params$bins)) %>%
+         group_by(quantile_index) %>%
+         summarize(x = median(x))
+       
+       # Get the number of each category in each bin 
+       counts <- data %>% mutate(quantile_index = dplyr::ntile(data$x, params$bins)) %>%
+         group_by(quantile_index, colour, PANEL, group) %>%
+         summarize(count = length(x))
+
+       # Combine the x and y data
+       data <- merge(median_x, counts, by = "quantile_index", all = TRUE)
+
+       # Now calculate the confidence intervals for the multinomial data
+       data <- data %>% group_by(quantile_index) %>%
+         mutate(x = median(x),
+                y=as.data.frame(DescTools::MultinomCI(count, params$conf_level))$est,
+                ymin=as.data.frame(DescTools::MultinomCI(count, params$conf_level))$lwr.ci,
+                ymax=as.data.frame(DescTools::MultinomCI(count, params$conf_level))$upr.ci) %>%
```

Comment:
Is there a way to use xgx_conf_int() to compute the y, ymin, and ymax lines here?

Reply:
Just a thought